creator-node/src/tracer.ts

```diff
@@ -65,6 +90,83 @@ export const setupTracing = () => {
   provider.register()
 }
 
+/**
+ * Higher-order function that adds opentelemetry tracing to a function.
+ * This wrapper works for both sync and async functions
+ *
+ * @param {string?} param.name optional name to give to the span, defaults to the function name
+ * @param {TFunction} param.fn the generic function to instrument
+ * @param {SpanOptions} param.options objects to pass into the span
+ * @returns the instrumented function
+ * @throws rethrows any errors from the original fn
+ *
+ * Usage of this would look like
+ * ```
+ * const someFunction = instrumentTracing({ fn: _someFunction })
+ * const result = someFunction(args))
+ * // or
+ * const result = await someFunction(args)
+ * ```
+ */
+export const instrumentTracing = <TFunction extends (...args: any[]) => any>({
+  name,
+  fn,
+  options
+}: {
+  name?: string
+  fn: TFunction
+  options?: SpanOptions
+}) => {
+  // build a wrapper around `fn` that accepts the same parameters and returns the same return type
+  const wrapper = function (
+    ...args: Parameters<TFunction>
+  ): ReturnType<TFunction> {
+    const spanName = name || fn.name
+    const spanOptions = options || {}
+    return tracing
+      .getTracer()
+      .startActiveSpan(spanName, spanOptions, (span: Span) => {
+        try {
+          tracing.setSpanAttribute(tracing.CODE_FUNCTION, fn.name)
+          tracing.setSpanAttribute('spid', SPID)
+          tracing.setSpanAttribute('endpoint', ENDPOINT)
+
+          // TODO add skip parameter to instrument testing function to NOT log certain args
+          // tracing.setSpanAttribute('args', JSON.stringify(args))
+          const result = fn.call(this, ...args)
```

Comment:
one more question about this - why use call and not apply?

Reply:
oh you spread which only call can do. apply takes in an array

Reply:
no reason - I just picked one

Is there a reason apply is better?

Reply:
It would just make that line look like
```js
const result = fn.apply(this, args)
```